Runtime/AvatarCreator/UI/Buttons/ButtonElement.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+    /// <summary>
+    /// A basic button element with an icon and functionality to bind an action to the
+    /// buttons onClick event.
+    /// </summary>
+    [RequireComponent(typeof(Button))]
+    public class ButtonElement : MonoBehaviour
```

Comment:
Is this component even needed? Why can't the avatar creator just expose a public variable to define a button prefab? The only thing this component does is enforce that the icon texture is set on the button, but even that is a bit too opinionated IMO. What if I want to go GTA style and have my buttons be the asset names, and not the icons?

![image](https://github.com/readyplayerme/rpm-unity-sdk-core/assets/121796327/fc77074e-7f3e-4802-a870-4aeb24162c15)

Reply:
valid point. I think that almost always when dealing with 2D UI you will have a Button component/element because generally to make something selectable or rather clickable it needs or be some sort of button. 
But you are right though they may want to have this component on some parent object or something and then be able to assign the button to something else in the inspector. 
The RequireComponent is sort of useless here anyways as I didn't find/assign it on start.

Reply:
To support text buttons we would probably need to also add a function to Set Text and also add a Text element 🤔

Reply:
leaving as is for now. If there is a need we can rework as needed based on feedback from devs